src/plugins/twitter/twitter.py

```diff
@@ -1,10 +1,135 @@
+                access_token_secret=current_app.config.get(
+                    'TWITTER_ACCESS_TOKEN_SECRET'
+                ),
+            )
+        except RuntimeError:
+            logger.error('App not found - skipping client instantiation')
+            return None
+
+    def _validate(self, results):
+        return self.validator.dump(results)
+
+    @staticmethod
+    def _convert_to_python(event_entry):
+        """ Parse data from twitter API and convert them to Python types."""
+        date_string = event_entry.pop('created_at_str')
+        created_at = datetime.strptime(date_string, '%a %b %d %X %z %Y')
+
+        twitter_id = event_entry.pop('twitter_id')
+        subject_id = set_generic_twitter_link(twitter_id)
+
+        return subject_id, created_at
+
+    def _build(self, event_data, uri_id, origin, event_dict):
+        """ Build a Event objects using the defined schema.
+
+        Args:
+            event_data (Iterable): list of Event dicts coming from the schema.
+            uri_id (int): id or uri being queried
+            origin (Enum): Service which originated the event we are fetching.
+            event_dict: dict of events not yet committed to the db
+
+        Returns:
+            tuple: The input event_dict and an Iterable of Event objects
+        """
+
+        events = {}
+        for event_entry in event_data:
+            subj, created_at = self._convert_to_python(event_entry)
+
+            event = self.get_event(
+                uri_id=uri_id,
+                subject_id=subj,
+                event_dict=event_dict
+            )
+
+            if not event:
+                event = Event(
+                    uri_id=uri_id,
+                    subject_id=subj,
+                    origin=origin.value,
+                    created_at=created_at
+                )
+                event_dict[subj] = event
+
+                events[event] = [
+                    RawEvent(
+                        event=event,
+                        scrape_id=event_entry['scrape_id'],
+                        origin=origin.value,
+                        provider=self.provider.value,
+                        created_at=created_at
+                    )
+                ]
+
+        return event_dict, events
+
+    def process(self, uri, origin, scrape, last_check, event_dict):
+        """ Implement processing of an URI to get Twitter events.
+
+        Args:
+            uri (Uri): An Uri object.
+            origin (Enum): Service which originated the event we are fetching.
+            scrape (Scrape): Scrape from ORM, not saved to database (yet).
+            last_check (datetime): when this uri was last successfully scraped.
+            event_dict: dict of events not yet committed to the db in the form:
+                {subj-id: event-object}
+
+        Returns:
+            generator: Contains events found for the given URI, as dictionaries.
```

Comment:
Is this supposed to return a python generator?? If so, I hate to tell you, it's not, unless I'm missing something in `_build` that creates a generator.

Reply:
It used to return a generator.... long ago. 
Thanks, will update the docstring :+1:

Reply:
Long ago, twas so long ago